code/controllers/subsystem/minimap.dm

```diff
@@ -338,6 +338,170 @@ SUBSYSTEM_DEF(minimaps)
+		return GLOB.uscm_unannounced_map
+	else if(faction == XENO_HIVE_NORMAL)
+		return GLOB.xeno_unannounced_map
+
+	return null
+
+/**
+ * Fetches the last set of svg coordinates for the tacmap drawing.
+ *
+ * Arguments:
+ * * faction: which faction get the map for: FACTION_MARINE or XENO_HIVE_NORMAL
+ */
+/proc/get_tacmap_data_svg(faction)
+	var/list/map_list
+
+	if(faction == FACTION_MARINE)
+		map_list = GLOB.uscm_svg_tacmap_data
+	else if(faction == XENO_HIVE_NORMAL)
+		map_list = GLOB.xeno_svg_tacmap_data
+	else
+		return null
+
+	var/map_length = length(map_list)
+
+	if(map_length == 0)
+		return null
+
+	return map_list[map_length]
+
+/**
+ * Sends relevant flattened tacmaps to a late joiner.
+ *
+ * Arguments:
+ * * user: The mob that is either an observer, marine, or xeno
+ */
+/proc/send_tacmap_assets_latejoin(mob/user)
+	if(!user.client)
+		return
+
+	var/is_observer = user.faction == FACTION_NEUTRAL && isobserver(user)
+	if(is_observer || user.faction == FACTION_MARINE)
+		// Send marine maps
+		var/datum/flattened_tacmap/latest = get_tacmap_data_png(FACTION_MARINE)
+		if(latest)
+			SSassets.transport.send_assets(user.client, latest.asset_key)
+		var/datum/flattened_tacmap/unannounced = get_unannounced_tacmap_data_png(FACTION_MARINE)
+		if(unannounced && (!latest || latest.asset_key != unannounced.asset_key))
+			SSassets.transport.send_assets(user.client, unannounced.asset_key)
+
+	var/mob/living/carbon/xenomorph/xeno = user
+	if(is_observer || istype(xeno) && xeno.hivenumber == XENO_HIVE_NORMAL)
+		// Send xeno maps
+		var/datum/flattened_tacmap/latest = get_tacmap_data_png(XENO_HIVE_NORMAL)
+		if(latest)
+			SSassets.transport.send_assets(user.client, latest.asset_key)
+		var/datum/flattened_tacmap/unannounced = get_unannounced_tacmap_data_png(XENO_HIVE_NORMAL)
+		if(unannounced && (!latest || latest.asset_key != unannounced.asset_key))
+			SSassets.transport.send_assets(user.client, unannounced.asset_key)
+
+/**
+ * Flattens the current map and then distributes it for the specified faction as an unannounced map.
+ *
+ * Arguments:
+ * * faction: which faction to distribute the map to: FACTION_MARINE or XENO_HIVE_NORMAL
+ * Return:
+ * * returns a boolean value, true if the operation was successful, false if it was not.
+ */
+/datum/tacmap/drawing/proc/distribute_current_map_png(faction)
+	if(faction == FACTION_MARINE)
+		if(!COOLDOWN_FINISHED(GLOB, uscm_flatten_map_icon_cooldown))
+			return FALSE
+		COOLDOWN_START(GLOB, uscm_flatten_map_icon_cooldown, flatten_map_cooldown_time)
+	else if(faction == XENO_HIVE_NORMAL)
+		if(!COOLDOWN_FINISHED(GLOB, xeno_flatten_map_icon_cooldown))
+			return FALSE
+		COOLDOWN_START(GLOB, xeno_flatten_map_icon_cooldown, flatten_map_cooldown_time)
+	else
+		return FALSE
+
+	var/icon/flat_map = getFlatIcon(map_holder.map, appearance_flags = TRUE)
+	if(!flat_map)
+		to_chat(usr, SPAN_WARNING("A critical error has occurred! Contact a coder.")) // tf2heavy: "Oh, this is bad!"
```

Comment:
This was originally more of an IC message that was changed to be what it is now because this shouldn't happen (nor have I seen this ever get hit) because I felt it made it seem like an intended problem and it is not an intended problem.